lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2095,6 +2095,59 @@ void SwiftLanguageRuntime::DidFinishExecutingUserExpression(
 
 bool SwiftLanguageRuntime::IsABIStable() { FORWARD(IsABIStable); }
 
+namespace {
+/// The target specific register numbers used for async unwinding.
+///
+/// For UnwindPlans, these use eh_frame / dwarf register numbering.
+struct AsyncUnwindRegisterNumbers {
+  uint32_t async_ctx_regnum;
+  uint32_t fp_regnum;
+  uint32_t pc_regnum;
+  /// A register to use as a marker to indicate how the async context is passed
+  /// to the function (indirectly, or not). This needs to be communicated to the
+  /// frames below us as they need to react differently. There is no good way to
+  /// expose this, so we set another dummy register to communicate this state.
+  uint32_t dummy_regnum;
+};
+} // namespace
+
+static llvm::Optional<AsyncUnwindRegisterNumbers>
+GetAsyncUnwindRegisterNumbers(llvm::Triple::ArchType triple) {
+  switch (triple) {
+  case llvm::Triple::x86_64:
+    return (AsyncUnwindRegisterNumbers){
```

Comment:
I think this syntax is C syntax that Clang might accept in C++, but it is not accepted by MSVC. Both VS2017 and VS2019 are failing in CI:

```
C:\jenkins\workspace\oss-swift-windows-x86_64\llvm-project\lldb\source\Target\SwiftLanguageRuntime.cpp(2119): error C2059: syntax error: '.'
C:\jenkins\workspace\oss-swift-windows-x86_64\llvm-project\lldb\source\Target\SwiftLanguageRuntime.cpp(2124): error C2046: illegal case
C:\jenkins\workspace\oss-swift-windows-x86_64\llvm-project\lldb\source\Target\SwiftLanguageRuntime.cpp(2126): error C2059: syntax error: '.'
```

and

```
S:\jenkins\workspace\oss-swift-windows-x86_64-vs2019\llvm-project\lldb\source\Target\SwiftLanguageRuntime.cpp(2119): error C7555: use of designated initializers requires at least '/std:c++latest'
S:\jenkins\workspace\oss-swift-windows-x86_64-vs2019\llvm-project\lldb\source\Target\SwiftLanguageRuntime.cpp(2123): error C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax
S:\jenkins\workspace\oss-swift-windows-x86_64-vs2019\llvm-project\lldb\source\Target\SwiftLanguageRuntime.cpp(2126): error C7555: use of designated initializers requires at least '/std:c++latest'
S:\jenkins\workspace\oss-swift-windows-x86_64-vs2019\llvm-project\lldb\source\Target\SwiftLanguageRuntime.cpp(2130): error C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax
```

I understand that this is more readable, but would anyone mind changing it to be standard C++? (I might had time to do it at some moment, but anyone feel free to change it before me)

Reply:
I'll check it thanks.

Reply:
https://github.com/apple/llvm-project/pull/2807

Reply:
Designated initializers are a part of C++20, so it's already part of the standard. Just not the standard we're using. :)